test/server/processOperations.test.ts

```diff
@@ -461,4 +461,242 @@ describe('processOperations', () => {
+
+        const operation3 = buildOperation(
+            reducer,
+            document,
+            actions.setModelId({ id: '4' }),
+            0
+        );
+
+        const resultOp3 = await server.addOperation(
+            driveId,
+            documentId,
+            operation3
+        );
+
+        expect(resultOp3.status).toBe('SUCCESS');
+
+        document = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
+
+        const operations = document.operations.global.slice(-4);
+
+        expect(document.state.global).toMatchObject({
+            name: '2',
+            id: '4'
+        });
+        expect(operations).toMatchObject([
+            {
+                type: 'SET_MODEL_NAME',
+                input: { name: '1' },
+                index: 6,
+                skip: 6
+            },
+            {
+                type: 'SET_MODEL_NAME',
+                input: { name: '2' },
+                index: 7,
+                skip: 0
+            },
+            {
+                type: 'SET_MODEL_ID',
+                input: { id: '3' },
+                index: 8,
+                skip: 0
+            },
+            {
+                type: 'SET_MODEL_ID',
+                input: { id: '4' },
+                index: 9,
+                skip: 0
+            }
+        ]);
+    });
+
+    it('should resolve conlficts suign the right order for merge oprations (simulate clients conflict)', async () => {
+        const initialDocument = await buildFile();
+        let pushOpreationResult: IOperationResult;
+
+        const client1 = new BasicClient(
+            server,
+            driveId,
+            documentId,
+            initialDocument,
+            reducer
+        );
+
+        const client2 = new BasicClient(
+            server,
+            driveId,
+            documentId,
+            initialDocument,
+            reducer
+        );
+
+        client1.dispatchDocumentAction(actions.setModelName({ name: '1' }));
+        pushOpreationResult = await client1.pushOperationsToServer();
+        expect(pushOpreationResult.status).toBe('SUCCESS');
+
+        client2.dispatchDocumentAction(actions.setModelName({ name: '2' }));
+        pushOpreationResult = await client2.pushOperationsToServer();
+        expect(pushOpreationResult.status).toBe('SUCCESS');
+
+        client2.dispatchDocumentAction(actions.setModelId({ id: '3' }));
+        pushOpreationResult = await client2.pushOperationsToServer();
+        expect(pushOpreationResult.status).toBe('SUCCESS');
+
+        client1.dispatchDocumentAction(actions.setModelId({ id: '4' }));
+        pushOpreationResult = await client1.pushOperationsToServer();
+        expect(pushOpreationResult.status).toBe('SUCCESS');
+
+        // Check the final state of the document
+        const finalDocument = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
```

Comment:
here there's an example of how we can use this new client to simulate different scenarios